src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_jni.h

```diff
@@ -465,6 +465,40 @@ extern jmethodID g_KeyAgreementInit;
 extern jmethodID g_KeyAgreementDoPhase;
 extern jmethodID g_KeyAgreementGenerateSecret;
 
+// Compatibility macros
+#if !defined (__mallocfunc)
+#if defined (__clang__) || defined (__GNUC__)
+#define __mallocfunc __attribute__((__malloc__))
+#else // def (__clang__ || __GNUC__)
+#define __mallocfunc
+#endif // ndef (__clang__ || __GNUC__)
+#endif
+
+#if !defined (__BIONIC_ALLOC_SIZE)
+#if defined (__clang__) || defined (__GNUC__)
+#define __BIONIC_ALLOC_SIZE(...) __attribute__((__alloc_size__(__VA_ARGS__)))
+#else // def (__clang__ || __GNUC__)
+#define __BIONIC_ALLOC_SIZE(...)
+#endif // ndef (__clang__ || __GNUC__)
+#endif
+
+#if !defined (__wur)
+#if defined (__clang__) || defined (__GNUC__)
+#define __wur __attribute__((__warn_unused_result__))
+#else // def (__clang__ || __GNUC__)
+#define __wur
+#endif // ndef (__clang__ || __GNUC__)
+#endif
+
+#if defined (__clang__) || defined (__GNUC__)
+#define NON_NULL(...) __attribute__((nonnull (__VA_ARGS__)))
+#else
+#define ALL_NON_NULL
+#define NON_NULL(_idx1_, ...)
```

Comment:
I'd prefer `ARGS_NON_NULL` and `ARGS_NON_NULL_ALL` to make it clear you're marking the function parameters.